src/Build/Globbing/MSBuildGlob.cs

```diff
@@ -145,7 +142,7 @@ private string NormalizeMatchInput(string stringToMatch)
 
             // Degenerate case when the string to match is empty.
             // Ensure trailing slash because the fixed directory part has a trailing slash.
-            if (stringToMatch == string.Empty)
+            if (string.IsNullOrEmpty(stringToMatch))
```

Comment:
this would allow white space to pass the test, right?

Reply:
No, `IsNullOrEmpty` really tests only for `null` and empty string. There's also `IsNullOrWhiteSpace` which is more permissive.